proxy/src/bind.rs

```diff
@@ -332,15 +331,25 @@ where
     }
 
     fn call(&mut self, mut request: S::Request) -> Self::Future {
-        if request.version() != http::Version::HTTP_2 {
-            h1::normalize_our_view_of_uri(&mut request, self.was_absolute_form);
+        if request.version() != http::Version::HTTP_2 &&
+            // Skip normalizing the URI if it was received in
+            // absolute form.
+            // TODO: we could probably skip this service altogether, if we
+            //       wrote some kind of `futures::Either` for services?
```

Comment:
i don't think this TODO is actionable in the scope of conduit. tower needs to take on the burden of supporting middlewares that are conditionally applied, but this does bad things to the type system. i suggest opening an issue on tower and removing the TODO here.